src/main/scala/group/scala/karazin/circe/literal/extras/macros.scala

```diff
@@ -149,21 +161,61 @@ object macros:
     
         case '[JsonObject] =>
           Json.fromJsonObject(JsonObjectUnit)
+
+        case '[Json] =>
+          JsonUnit
+
+        case '[JsonNumber] =>
+          Json.fromJsonNumber(JsonNumberUnit)
     
         case '[t] if TypeRepr.of[t] <:< TypeRepr.of[Product] =>
-          Json.fromFields(deconstructArgumentProduct[t])      
+          Json.fromFields(deconstructArgumentProduct[t])    
     
-        case '[Boolean] => 
+        case '[Boolean] | '[java.lang.Boolean] =>
           Json.fromBoolean(BooleanUnit)
+
+        case '[Byte] | '[java.lang.Byte] =>
+          Json.fromInt(ByteUnit.toInt)
```

Comment:
You don't need to cast the values.
Review dashed lines. It means that Byte can be treated as Int without any conversion. Fix it everywhere.

![image](https://user-images.githubusercontent.com/12128481/116525918-b1db9d80-a8e1-11eb-8458-a09c460f0638.png)